Router unable to route request to do_DELETE_Attachment

test https://github.com/couchbaselabs/cblite-tests/blob/master/tests/cbl-document.js#L270


manually:
```
curl http://localhost:8081/cbl-document2/cbl-document2_0
{"at":"2014-05-05T12:15:43.984Z","text":"Inline text string created by cblite functional test","_rev":"4-e9c315d9e365605fcb2cca5974fb26c4","_id":"cbl-document2_0","_attachments":{"ggate.png":{"length":210359,"digest":"sha1-/9+Ffm0wXDLzDjrqtLhW0VIVDHI=","revpos":1,"content_type":"image/png","stub":true},"inline.txt":{"length":52,"digest":"sha1-i/STphkUNnLhfyegJkGZIcUECYk=","revpos":4,"content_type":"text/plain","stub":true}}}

curl -X DELETE http://localhost:8081/cbl-document2/cbl-document2_0/ggate.png?rev=4-e9c315d9e365605fcb2cca5974fb26c4
{"error":"not_found","reason":"Router unable to route request to do_DELETE_Attachmentjava.lang.reflect.InvocationTargetException"}
```

logcat:
```
V/Listener( 1191): Not enforcing basic auth -- allowedCredentials null or empty
E/Router  ( 1191): Router unable to route request to do_DELETE_Attachment
E/Router  ( 1191): java.lang.reflect.InvocationTargetException
E/Router  ( 1191): 	at java.lang.reflect.Method.invokeNative(Native Method)
E/Router  ( 1191): 	at java.lang.reflect.Method.invoke(Method.java:515)
E/Router  ( 1191): 	at com.couchbase.lite.router.Router.start(Router.java:464)
E/Router  ( 1191): 	at com.couchbase.lite.listener.LiteServlet.service(LiteServlet.java:132)
E/Router  ( 1191): 	at javax.servlet.http.HttpServlet.service(HttpServlet.java:802)
E/Router  ( 1191): 	at Acme.Serve.Serve$ServeConnection.runServlet(Serve.java:2251)
E/Router  ( 1191): 	at Acme.Serve.Serve$ServeConnection.parseRequest(Serve.java:2176)
E/Router  ( 1191): 	at Acme.Serve.Serve$ServeConnection.run(Serve.java:1988)
E/Router  ( 1191): 	at Acme.Utils$ThreadPool$PooledThread.run(Utils.java:1223)
E/Router  ( 1191): 	at java.lang.Thread.run(Thread.java:841)
E/Router  ( 1191): Caused by: java.lang.NullPointerException
E/Router  ( 1191): 	at com.couchbase.lite.util.StreamUtils.copyStream(StreamUtils.java:27)
E/Router  ( 1191): 	at com.couchbase.lite.router.Router.updateAttachment(Router.java:1583)
E/Router  ( 1191): 	at com.couchbase.lite.router.Router.do_DELETE_Attachment(Router.java:1609)
E/Router  ( 1191): 	... 10 more
```

--- a/src/main/java/com/couchbase/lite/router/Router.java
+++ b/src/main/java/com/couchbase/lite/router/Router.java
@@ -2198,25 +2198,38 @@ public class Router implements Database.ChangeListener, Database.DatabaseListene
         return update(_db, docID, null, true);
     }
 
+    /**
+     * @param contentStream if null, delete attachment. if not null, update attachment
+     */
     private Status updateAttachment(String attachment,
                                     String docID,
                                     InputStream contentStream)
             throws CouchbaseLiteException {
         Status status = new Status(Status.OK);
         String revID = getQuery("rev");
-        if (revID == null) {
+        if (revID == null)
             revID = getRevIDFromIfMatchHeader();
-        }
 
-        BlobStoreWriter body = new BlobStoreWriter(db.getAttachmentStore());
-        ByteArrayOutputStream dataStream = new ByteArrayOutputStream();
+        if (revID == null || revID.length() == 0)
+            throw new CouchbaseLiteException(Status.BAD_REQUEST);
 
-        try {
-            StreamUtils.copyStream(contentStream, dataStream);
-            body.appendData(dataStream.toByteArray());
-            body.finish();
-        } catch (Exception e) {
-            throw new CouchbaseLiteException(e.getCause(), Status.BAD_ATTACHMENT);
+        BlobStoreWriter body = null;
+        if (contentStream != null) {
+            body = new BlobStoreWriter(db.getAttachmentStore());
+            ByteArrayOutputStream dataStream = new ByteArrayOutputStream();
+            try {
+                StreamUtils.copyStream(contentStream, dataStream);
+                body.appendData(dataStream.toByteArray());
+                body.finish();
+            } catch (Exception e) {
+                throw new CouchbaseLiteException(e.getCause(), Status.BAD_ATTACHMENT);
+            } finally {
+                try {
+                    dataStream.close();
+                } catch (IOException e) {
+                    throw new CouchbaseLiteException(e.getCause(), Status.BAD_ATTACHMENT);
+                }
+            }
         }
 
         // updateAttachment uses transaction internally, not necessary to be synchronized

